src/node/chainstatemanager_args.cpp

```diff
@@ -39,7 +39,15 @@ util::Result<void> ApplyArgsManOptions(const ArgsManager& args, ChainstateManage
         }
     }
 
-    if (auto value{args.GetArg("-assumevalid")}) opts.assumed_valid_block = uint256S(*value);
+    if (auto value{args.GetArg("-assumevalid")}) {
+        if (*value == "0") { // handle -noassumevalid
+            opts.assumed_valid_block = uint256{};
+        } else if (auto block_hash{uint256::FromHex(*value)}) {
```

Comment:
I didn't previously realize that this PR breaks use-facing compatibility wrt `0x` prefixes in args. Would also like to see such cases be filtered through `TrySanitizeHexNumber` before going into `FromHex` so that level of compatibility is maintained.

Reply:
Thank you for summarizing your concerns @ryanofsky. I believe the latest force-push addresses all of them.

> It's good this change is making it an error to include trailing non-hex characters instead of silently ignoring them.
It's good this change is making it an error to specify hex strings that are too long.
It's probably ok this change makes it an error to include leading and trailing whitespace, instead of trimming it. Maybe that could be safer and prevent configuration bugs.

Relevant behaviour unchanged in latest force-push.

> It's not good but probably ok this change makes it an error to specify an empty string (-assumedvalid="") to disable the assumevalid feature. That seemed like a useful way to clear the setting.
It's not good this change makes it an error to include a 0x prefix or write numbers not padded with 0's. That just seems like being arbitrarily strict and breaking backwards compatibility for no reason.

Latest force-push continues to allow `-assumevalid=""` as well as 0x-prefixed and too-short-input (and unit test cases are added to `test: unittest chainstatemanager_args`), keeping this behaviour the same as it currently is on master.